integreat_chat/chatanswers/services/answer_service.py

```diff
@@ -98,3 +105,28 @@ def extract_answer(self, question):
             "sources": list({result[0].metadata["source"] for result in results}),
             "details": self.doc_details(results)
         }
+
+
+    def check_document_relevance(self, question, content):
+        """
+        Check if the retrieved documents are relevant
+        """
+
+        system = """You are a grader assessing relevance of a retrieved document to a user question. \n 
+        If the document contains keyword(s) or semantic meaning related to the user question, grade it as relevant. \n
+        It does not need to be a stringent test. The goal is to filter out erroneous retrievals. \n
+        Give a binary score 'yes' or 'no' score to indicate whether the document is relevant to the question and only answer with either 'yes' or 'no'."""
+        
+        grade_prompt = ChatPromptTemplate.from_messages(
+            [
+                ("system", system),
+                ("human", "Retrieved document: \n\n {document} \n\n User question: {question}"),
+            ]
+        )
+        chain = grade_prompt | self.llm | StrOutputParser()
+        
+        response = chain.invoke({"document": content, "question": question})
+        response = response.strip().lower()
+        print(content, '\n', '-'*25)
```

Comment:
Except for the `print()` statements, this looks good :tada: Thanks!

Reply:
Oh yes. Changed now:) Thanks